packages/bsky/src/services/feed/index.ts

```diff
@@ -280,29 +204,51 @@ export class FeedService {
         actorDids.add(new AtUri(item.replyRoot).hostname)
       }
     }
-    const [actors, posts, labels] = await Promise.all([
-      this.services.actor.views.profiles(Array.from(actorDids), viewer, {
-        skipLabels: true,
-      }),
-      this.getPostInfos(Array.from(postUris), viewer),
-      this.services.label.getLabelsForSubjects([...postUris, ...actorDids]),
-    ])
-    const blocks = await this.blocksForPosts(posts)
-    const embeds = await this.embedsForPosts(posts, blocks, viewer)
+    return { dids: actorDids, uris: postUris }
+  }
 
-    return this.views.formatFeed(
-      items,
-      actors,
+  async feedHydration(
```

Comment:
not to muck this up even more, but we want to treat block violations different in different contexts. for instance, a block violating reply is:
- not shown in thread
- shown in author feed

block violating quote posts are still shown but detach the quoted post

Reply:
Fixing that behavior seems like a good chunk of work on which to hinge some further dismantling of feed hydration.  I am pretty sure it would be "additive" onto the work done so far here, so I think we have leeway about whether to bite it off now or handle separately.

Reply:
yeah makes sense - i'm down to tackle it as a separate chunk of work 👌